qrenderdoc/Windows/PipelineState/D3D12PipelineStateViewer.cpp

```diff
@@ -800,19 +799,15 @@ bool D3D12PipelineStateViewer::showNode(bool usedSlot, bool filledSlot)
   const bool showUnused = ui->showUnused->isChecked();
   const bool showEmpty = ui->showEmpty->isChecked();
 
-  // show if it's referenced by the shader - regardless of empty or not
-  if(usedSlot)
-    return true;
-
-  // it's not referenced, but if it's bound and we have "show unused" then show it
-  if(showUnused && filledSlot)
-    return true;
+  // It's empty, and we don't have "show empty" checked
```

Comment:
This logic should not change as part of this PR, it's not needed for the root signature re-organisation. The pipeline viewer rules are consistent across different APIs - if it's used by the shader, it's always displayed regardless of whether it's empty or not.